backend/de.metas.contracts/src/main/java/de/metas/contracts/bpartner/service/OrgChangeCommand.java

```diff
@@ -764,36 +764,63 @@ private void saveOrgChangeBPartnerComposite(@NonNull final OrgChangeBPartnerComp
 		bpCompositeRepo.save(bPartnerComposite, false);
 	}
 
-	private void createOrgSwitchRequest(@NonNull final OrgChangeHistoryId orgChangeHistoryId)
+	private void createOrgSwitchRequests(@NonNull final OrgChangeHistoryId orgChangeHistoryId)
 	{
 		final RequestTypeId requestTypeId = requestTypeDAO.retrieveOrgChangeRequestTypeId();
 		final I_AD_OrgChange_History orgChangeHistoryRecord = orgChangeHistoryRepo.getOrgChangeHistoryById(orgChangeHistoryId);
 
-		final OrgId orgId = OrgId.ofRepoId(orgChangeHistoryRecord.getAD_OrgTo_ID());
+		final OrgId orgToId = OrgId.ofRepoId(orgChangeHistoryRecord.getAD_OrgTo_ID());
 
-		final ZoneId timeZone = orgDAO.getTimeZone(orgId);
-		final ZonedDateTime requestDate = TimeUtil.asZonedDateTime(orgChangeHistoryRecord.getDate_OrgChange(), timeZone);
+		final ZoneId orgToTimeZone = orgDAO.getTimeZone(orgToId);
+		final ZonedDateTime orgToRequestDate = TimeUtil.asZonedDateTime(orgChangeHistoryRecord.getDate_OrgChange(), orgToTimeZone);
 
-		final BPartnerId bPartnerId = BPartnerId.ofRepoId(orgChangeHistoryRecord.getC_BPartner_To_ID());
+		final BPartnerId orgToBPartnerId = BPartnerId.ofRepoId(orgChangeHistoryRecord.getC_BPartner_To_ID());
 
-		final String summary = msgBL.getMsg(Env.getCtx(), MSG_OrgChangeSummary, new Object[] {
-				orgDAO.getById(orgChangeHistoryRecord.getAD_Org_From_ID()).getName(),
-				orgDAO.getById(orgChangeHistoryRecord.getAD_OrgTo_ID()).getName(),
-				TimeUtil.asDate(requestDate) });
+		final String orgToSummary = getSummaryForOrg(orgChangeHistoryRecord, orgToRequestDate);
 
-		final RequestCandidate requestCandidate = RequestCandidate.builder()
-				.summary(summary)
+		final RequestCandidate orgToRequestCandidate = RequestCandidate.builder()
+				.summary(orgToSummary)
 				.confidentialType(X_R_Request.CONFIDENTIALTYPE_PartnerConfidential)
-				.orgId(orgId)
-				.recordRef(TableRecordReference.of(I_C_BPartner.Table_Name, bPartnerId))
+				.orgId(orgToId)
+				.recordRef(TableRecordReference.of(I_C_BPartner.Table_Name, orgToBPartnerId))
 				.requestTypeId(requestTypeId)
-				.partnerId(bPartnerId)
-				.dateDelivered(requestDate)
+				.partnerId(orgToBPartnerId)
+				.dateDelivered(orgToRequestDate)
+				.build();
+
+		requestBL.createRequest(orgToRequestCandidate);
+
+		final OrgId orgFromId = OrgId.ofRepoId(orgChangeHistoryRecord.getAD_Org_From_ID());
```

Comment:
Since most of this is copy-paste from above, why not extract to a `createOrgSwitchRequest(OrgId, I_AD_OrgChange_History)` method, and reuse it for both requests?

Reply:
The function would need more parameters, because of different Org and bPartner are needed from `I_AD_OrgChange_History` and with maybe additional distinctions later on i decided against extracting